src/DotNetLightning.Core/Channel/ChannelHelpers.fs

```diff
@@ -17,3 +24,431 @@ module internal ChannelConstantHelpers =
+
+                let perCommitmentPoint = perCommitmentSecret.PerCommitmentPoint()
+
+                let localCommitmentPubKeys =
+                    perCommitmentPoint.DeriveCommitmentPubKeys localChannelPubKeys
+
+                let remoteCommitmentPubKeys =
+                    perCommitmentPoint.DeriveCommitmentPubKeys remoteChannelPubKeys
+
+                let transactionBuilder =
+                    staticChannelConfig.Network.CreateTransactionBuilder()
+
+                let toRemoteScriptPubKey =
+                    localCommitmentPubKeys
+                        .PaymentPubKey
+                        .RawPubKey()
+                        .WitHash
+                        .ScriptPubKey
+
+                let toLocalScriptPubKey =
+                    Scripts.toLocalDelayed
+                        localCommitmentPubKeys.RevocationPubKey
+                        staticChannelConfig.LocalParams.ToSelfDelay
+                        remoteCommitmentPubKeys.DelayedPaymentPubKey
+
+                let toLocalWitScriptPubKey = toLocalScriptPubKey.WitHash.ScriptPubKey
+
+                let toRemoteIndexOpt =
+                    closingTx.Outputs
+                    |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
+
+                toRemoteIndexOpt
+                |> Option.iter
+                    (
+                        fun toRemoteIndex ->
+                            let localPaymentPrivKey =
+                                perCommitmentPoint.DerivePaymentPrivKey
+                                    localChannelPrivKeys.PaymentBasepointSecret
+
+                            transactionBuilder.SetVersion TxVersionNumberOfCommitmentTxs
+                            |> ignore<TransactionBuilder>
+
+                            transactionBuilder.AddKeys(localPaymentPrivKey.RawKey())
+                            |> ignore<TransactionBuilder>
+
+                            transactionBuilder.AddCoin(Coin(closingTx, toRemoteIndex |> uint32)) |> ignore
+                    )
+
+                let toLocalIndexOpt =
+                    closingTx.Outputs
+                    |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toLocalWitScriptPubKey)
+
+                toLocalIndexOpt
+                |> Option.iter
+                    (fun toLocalIndex ->
+                        let revocationPrivKey =
+                            perCommitmentSecret.DeriveRevocationPrivKey localChannelPrivKeys.RevocationBasepointSecret
+
+                        transactionBuilder.Extensions.Add(CommitmentToLocalExtension())
+
+                        transactionBuilder
+                            .AddKeys(revocationPrivKey.RawKey())
+                            .AddCoin(ScriptCoin(closingTx, toLocalIndex |> uint32, toLocalScriptPubKey))
+                        |> ignore)
+
+                // FIXME: what if all the money is in htlcs?
+                if toLocalIndexOpt.IsNone && toRemoteIndexOpt.IsNone then
```

Comment:
This is a temporary fix for the false-positive scenario, this causes the function to return "Unknown Closing Tx" if we couldn't find any to_local and to_remote output, but this creates an issue with commitment TXs that only have HTLC outputs and not any main one.

Reply:
I want you to add test to check the false positive scenario.

>  but this creates an issue with commitment TXs that only have HTLC outputs and not any main one.

And a test to check that this won't lead to the crash?

Reply:
> I want you to add test to check the false positive scenario.
I added a test for old local commitment txs 
> And a test to check that this won't lead to the crash?
I think I should explain more, the only problem with that edge case is that instead of BalanceBelowDustLimit error we return UnknownClosingTx which seems a bad error for the case (I added a comment explaining it the code also)